src/lib/utils/__tests__/mergeSearchParameters-test.ts

```diff
@@ -0,0 +1,370 @@
+import algoliasearchHelper from 'algoliasearch-helper';
+import merge from '../mergeSearchParameters';
+
+describe('mergeSearchParameters', () => {
+  it('merges non-managed parameters', () => {
+    const actual = merge(
+      algoliasearchHelper.SearchParameters.make({
+        // Inherit
+        hitsPerPage: 2,
+        attributesToSnippet: ['description'],
+        // Overridden
+        query: 'Samsung',
+        attributesToHighlight: ['name'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        // hitsPerPage: 2,
+        // attributesToSnippet: ['description'],
+        query: 'Apple',
+        attributesToHighlight: ['name', 'author'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        // hitsPerPage: 2,
+        // attributesToSnippet: ['description'],
+        // attributesToHighlight: ['name', 'author'],
+        query: 'Apple iPhone',
+        distinct: true,
+      })
+    );
+
+    expect(actual).toEqual(
+      algoliasearchHelper.SearchParameters.make({
+        hitsPerPage: 2,
+        attributesToSnippet: ['description'],
+        attributesToHighlight: ['name', 'author'],
+        query: 'Apple iPhone',
+        distinct: true,
+      })
+    );
+  });
+
+  it('merges `facets` parameters', () => {
+    const actual = merge(
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['categories'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'colors'],
+      })
+    );
+
+    expect(actual).toEqual(
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'categories', 'colors'],
+      })
+    );
+  });
+
+  it('merges `disjunctiveFacets` parameters', () => {
+    const actual = merge(
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['brand'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['categories'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['brand', 'colors'],
+      })
+    );
+
+    expect(actual).toEqual(
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['brand', 'categories', 'colors'],
+      })
+    );
+  });
+
+  it('merges `facetsRefinements` parameters', () => {
```

Comment:
```suggestion
  it('merges `facetsRefinements` parameters, overrides conflicts', () => {
```

I think a title like this would explain more the behaviour we want. It could also be split up into separate tests